substrate/client/authority-discovery/src/worker.rs

```diff
@@ -126,7 +126,9 @@ pub struct Worker<Client, Network, Block, DhtEventStream> {
 	/// List of keys onto which addresses have been published at the latest publication.
 	/// Used to check whether they have changed.
 	latest_published_keys: HashSet<AuthorityId>,
-	/// Same value as in the configuration.
+	/// Should we publish only these explicitly set addresses?
+	public_addresses_only: Option<Vec<Multiaddr>>,
```

Comment:
nit: We could prob have an enum for the mutual exclusive flags. Thinking outloud here, totally up to you

```rust
enum PublishedAddresses {
    List(Vec<Multiaddr>),
    External { non_global: bool }
}
```

Reply:
Seems reasonable, done.